File: Java/setZeroes.java
public class Solution {
    public void setZeroes(int[][] matrix) {
        // Start typing your Java solution below
        // DO NOT write main() function
        int ROW = matrix.length;
        int COL = matrix[0].length;
        boolean[] row = new boolean[ROW];
        boolean[] col = new boolean[COL];
        
        for (int i = 0; i < ROW; i++)
        {
            for (int j = 0; j < COL; j++)
            {
                if (matrix[i][j] == 0)
                {
                    row[i] = true;
                    col[j] = true;
                }
            }
        }
        
        for (int i = 0; i < ROW; i++)
        {
            for (int j = 0; j < COL; j++)
            {
                if (row[i] || col[j])
                {
                    matrix[i][j] = 0;
                }
            }
        }
        
    }
}

File: C++/setZeroes.cc
void setZeroes(vector<vector<int> > &matrix) 
{
	// Start typing your C/C++ solution below
	// DO NOT write int main() function
	int row = matrix.size();
	int col = matrix[0].size();
	bool *zeroRow = new bool[row];
	bool *zeroCol = new bool[col];
	memset(zeroRow, false, row);
	memset(zeroCol, false, col);
	
	for (int i = 0; i < row; i++)
	{
		for (int j = 0; j < col; j++)
		{
			if (matrix[i][j] == 0)
			{
				zeroRow[i] = true;
				zeroCol[j] = true;
			}
		}
	}
	
	for (int i = 0; i < row; i++)
	{
		for (int j = 0; j < col; j++)
		{
			if (zeroRow[i] == true || zeroCol[j] == true)
			{
				matrix[i][j] = 0;
			}
		}
	}
}
